src/types/bloom_filter.h

```diff
@@ -46,6 +49,40 @@ constexpr bool IsMultipleOf8(int64_t n) { return (n & 7) == 0; }
 // This value will be reconsidered when implementing Bloom filter producer.
 static constexpr uint32_t kMaximumBloomFilterBytes = 128 * 1024 * 1024;
 
+/// Minimum Bloom filter size, it sets to 32 bytes to fit a tiny Bloom filter.
+static constexpr uint32_t kMinimumBloomFilterBytes = 32;
+
+class BlockSplitBloomFilter;
+
+using OwnedBlockSplitBloomFilter = std::tuple<BlockSplitBloomFilter, std::string>;
+
+/// Initialize the BlockSplitBloomFilter. The range of num_bytes should be within
+/// [kMinimumBloomFilterBytes, kMaximumBloomFilterBytes], it will be
+/// rounded up/down to lower/upper bound if num_bytes is out of range and also
+/// will be rounded up to a power of 2.
+///
+/// @param num_bytes The number of bytes to store Bloom filter bitset.
+OwnedBlockSplitBloomFilter CreateBlockSplitBloomFilter(uint32_t num_bytes);
+
+/// Initialize the BlockSplitBloomFilter. It copies the bitset as underlying
+/// bitset when the given bitset may not satisfy the 32-byte alignment requirement
+/// which may lead to segfault when performing SIMD instructions. It is the caller's
+/// responsibility to free the bitset passed in.
```

Comment:
Maybe we should modify some comments like this?
```suggestion
/// responsibility to ensure the bitset would not be changed when using the bloomfilter.
```

Reply:
the bitset can be changed via `InsertHash`.

Reply:
I guess @zncleon means the mutability. Sometime we may need `const` ctor

Reply:
it seems not related to this function

Reply:
Yeah, I mean can we provide something like:

```
StatusOr<unique_ptr<BlockSplitBloomFilter>> CreateBlockSplitBloomFilter(uint8_t* bitset, uint32_t num_bytes);
StatusOr<unique_ptr<BlockSplitBloomFilter>> CreateBlockSplitBloomFilter(std::string& bitset);
StatusOr<unique_ptr<const BlockSplitBloomFilter>> CreateBlockSplitBloomFilter(const uint8_t* bitset, uint32_t num_bytes);
StatusOr<unique_ptr<const BlockSplitBloomFilter>> CreateBlockSplitBloomFilter(const std::string& bitset);
```

Reply:
if a readonly version is needed, you can have a `const BlockSplitBloomFilter`.

Reply:
Got your point. I think you can add it when you need it.